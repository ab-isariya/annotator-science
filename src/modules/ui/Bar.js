import PropTypes from 'prop-types';

/**
 * Horizontal background bar
 *
 * @param {String} bgColor - background color (tailwind valid)
 * @param {String} padding - (tailwind valid)
 * @param {String} size - width of background
 * @param {Object} children - React elements to appear on top of bar
 * @return {JSX.Element}
 */
const Bar = ({bgColor, padding, size, children}) => (
  <div className="relative flex-1">
    {/* Background bar */}
    <div
      className={`absolute h-full left-0 rounded-r-md ${bgColor}`}
      style={{width: `${size}%`}}
    />
    <div className={`relative flex items-center ${padding}`}>{children}</div>
  </div>
);

Bar.propTypes = {
  //Color of the background horizontal bar
  bgColor: PropTypes.string.isRequired,
  //Padding class names from tailwind for the horizontal bar
  padding: PropTypes.string,
  //Percentage of the bar to be filled (0% - 100%)
  size: PropTypes.number.isRequired,
  //Children nodes to be rendered inside of the bar.
  children: PropTypes.node
};

export default Bar;
