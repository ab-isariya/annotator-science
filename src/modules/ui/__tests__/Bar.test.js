import Bar from '../Bar';
import renderer from 'react-test-renderer';

describe('<Bar/>', () => {
  it('Renders without crashing', () => {
    const comp = renderer
      .create(<Bar bgColor="bg-blue-500" size={0} />)
      .toJSON();

    expect(comp).toMatchSnapshot();
  });

  const Wrapper = ({children, width}) => <div style={{width}}>{children}</div>;

  it('Renders with a blue-500 bar at 100% width', () => {
    const bar = <Bar bgColor="bg-blue-500" size={100} />;

    const comp = renderer
      .create(<Wrapper width="200px">{bar}</Wrapper>)
      .toJSON();

    expect(comp.children[0].children[0].props.className).toMatch(
      /(bg-blue-500)/i
    );
    expect(comp.children[0].children[0].props.style.width).toMatch(/(100%)/i);
    expect(comp).toMatchSnapshot();
  });

  it('Renders with children, a red-500 bar, and a padding of px-2', () => {
    const comp = renderer
      .create(
        <Bar bgColor="bg-red-500" size={100} padding="px-2">
          I should render.
        </Bar>
      )
      .toJSON();

    expect(comp.children[0].props.className).toMatch(/(bg-red-500)/i);
    expect(comp.children[1].props.className).toMatch(/(px-2)/i);
    expect(comp).toMatchSnapshot();
  });
});
