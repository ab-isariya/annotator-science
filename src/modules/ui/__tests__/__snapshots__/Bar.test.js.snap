// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bar/> Renders with a blue-500 bar at 100% width 1`] = `
<div
  style={
    Object {
      "width": "200px",
    }
  }
>
  <div
    className="relative flex-1"
  >
    <div
      className="absolute h-full left-0 rounded-r-md bg-blue-500"
      style={
        Object {
          "width": "100%",
        }
      }
    />
    <div
      className="relative flex items-center undefined"
    />
  </div>
</div>
`;

exports[`<Bar/> Renders with children, a red-500 bar, and a padding of px-2 1`] = `
<div
  className="relative flex-1"
>
  <div
    className="absolute h-full left-0 rounded-r-md bg-red-500"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    className="relative flex items-center px-2"
  >
    I should render.
  </div>
</div>
`;

exports[`<Bar/> Renders without crashing 1`] = `
<div
  className="relative flex-1"
>
  <div
    className="absolute h-full left-0 rounded-r-md bg-blue-500"
    style={
      Object {
        "width": "0%",
      }
    }
  />
  <div
    className="relative flex items-center undefined"
  />
</div>
`;
